### DISCUSSION TITLE
Prevention of MySQL injection

### DISCUSSION THREAD

**Discussion by CodeGitOverflow** [community/community#76780] [Programming Help]
### Body

What are MySQL injections and how to prevent them?

Also,
As you can tell I have an extremely low IQ and comprehension, so if possible please make it over simplified for my dumbass.

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by melvincwng** [comment]
@CodeGitOverflow, in summary:

> - SQL injections are an attacking mechanism in which attackers use malicious SQL queries to cause damage to your backend databases (e.g. dropping the entire database)
> - You can prevent SQL injections via a variety of ways - most common methods are parameterized queries and input sanitization
> - There are many useful resources that can simplify the explanation of SQL injections & its prevention. For example: [this](https://www.youtube.com/watch?v=Yqu93GXx0vI) and [this](https://www.codecademy.com/learn/seasp-defending-node-applications-from-sql-injection-xss-csrf-attacks/modules/seasp-preventing-sql-injection-attacks/cheatsheet)

Hope this helps and answers your question 😄

**Comment by davea42** [comment]
For example, if one allows users to type just anything in a field and you use that in an sql command,
 if the user typed  
  `  ;  drop table photos ;`  
the entire photos table would  vanish.
Even if there was more than just that in the sql command.  
Because the sql command parser is happy to do things that
only the database administrator should actually do.

A typical way to avoid such disasters is to
scan the user typed string and delete dangerous characters from the string (for sure delete any semicolon,
and maybe also  other characters used in sql commands) before sending the string to the sql command parser.
Consider carefuly what you allow in user-typed strings and  only allow things that are important,
rather than allowing just anything.
